app/controllers/triannon/annotations_controller.rb

```diff
@@ -126,13 +133,13 @@ def create
     # DELETE /annotations/1
     def destroy
       @annotation.destroy
-      redirect_to annotations_url, status: 204, notice: 'Annotation was successfully destroyed.'
+      redirect_to "/annotations/#{params[:anno_root]}", status: 204, notice: 'Annotation was successfully destroyed.'
     end
 
 private
 
     def set_annotation
-      @annotation = Annotation.find(params[:id])
+      @annotation = Annotation.find(params[:id], params[:anno_root])
```

Comment:
Do you always have an `anno_root`? If so, maybe `Annotation.find(anno_root, id)` would reflect the hierarchy a little better?